src/utils/data-formatters/schema/format-history-pending-event-schema.ts

```diff
@@ -0,0 +1,44 @@
+import { type z } from 'zod';
+
+import { type PendingHistoryEvent } from '@/views/workflow-history/workflow-history.types';
+
+import formatPendingActivityTaskStartEvent from '../format-pending-workflow-history-event/format-pending-activity-start-event';
+import formatPendingDecisionTaskScheduleEvent from '../format-pending-workflow-history-event/format-pending-decision-schedule-event';
+
+import {
+  pendingActivityTaskStartSchema,
+  pendingDecisionTaskScheduleSchema,
+} from './pending-history-event-schema';
+
+export const formatPendingActivityTaskStartEventSchema =
+  pendingActivityTaskStartSchema.transform(formatPendingActivityTaskStartEvent);
+
+export const formatPendingDecisionTaskScheduleEventSchema =
+  pendingDecisionTaskScheduleSchema.transform(
+    formatPendingDecisionTaskScheduleEvent
+  );
+
+function unExistingEventType(_: never) {
+  return null;
+}
+export const getFormatPendingEventSchema = (event: PendingHistoryEvent) => {
+  switch (event.attributes) {
+    case 'pendingActivityTaskStartEventAttributes':
+      return formatPendingActivityTaskStartEventSchema;
+    case 'pendingDecisionTaskScheduleEventAttributes':
+      return formatPendingDecisionTaskScheduleEventSchema;
+    default:
+      return unExistingEventType(event); // should not be unreachable, used to show a type error if not all attributes cases are covered
```

Comment:
The intent of this comment can be clarified better
```suggestion
      return unExistingEventType(event); // used to show a type error if any pending event attributes cases are not covered
```